### PR TITLE
Fixes teleport and syzkaller builds

### PR DESCRIPTION
**Pull request by catenacyber**
By running go mod vendor after go mod tidy

cc @inferno-chromium more go 1.16

cc @mvdan following our discussion in another issue...
cc @awly for the fix about teleport

### CODE REVIEW

**Review by inferno-chromium — APPROVED** [review]

**Comment by inferno-chromium** [comment]
Can you check the failures
```
+ go mod vendor
go: finding module for package github.com/google/syzkaller/sys/akaros/gen
go: finding module for package github.com/google/syzkaller/sys/test/gen
go: finding module for package github.com/google/syzkaller/sys/freebsd/gen
go: finding module for package github.com/google/syzkaller/sys/fuchsia/gen
go: finding module for package github.com/google/syzkaller/sys/linux/gen
go: finding module for package github.com/google/syzkaller/sys/netbsd/gen
go: finding module for package github.com/google/syzkaller/sys/openbsd/gen
go: finding module for package github.com/google/syzkaller/sys/windows/gen
github.com/google/syzkaller/prog/test imports
	github.com/google/syzkaller/sys/test/gen: no matching versions for query "latest"
github.com/google/syzkaller/sys imports
	github.com/google/syzkaller/sys/akaros/gen: no matching versions for query "latest"
github.com/google/syzkaller/sys imports
	github.com/google/syzkaller/sys/freebsd/gen: no matching versions for query "latest"
github.com/google/syzkaller/sys imports
	github.com/google/syzkaller/sys/fuchsia/gen: no matching versions for query "latest"
github.com/google/syzkaller/sys imports
	github.com/google/syzkaller/sys/linux/gen: no matching versions for query "latest"
github.com/google/syzkaller/sys imports
	github.com/google/syzkaller/sys/netbsd/gen: no matching versions for query "latest"
github.com/google/syzkaller/sys imports
	github.com/google/syzkaller/sys/openbsd/gen: no matching versions for query "latest"
github.com/google/syzkaller/sys imports
	github.com/google/syzkaller/sys/windows/gen: no matching versions for query "latest"
Building fuzzers failed.
````

File: projects/teleport/build.sh [hunk 1]
```diff
@@ -15,10 +15,7 @@
 #
 ################################################################################
 
-
-mkdir -p $GOPATH/src/github.com/gravitational
-cd $GOPATH/src/github.com/gravitational
-git clone https://github.com/gravitational/teleport.git
+go mod tidy && go mod vendor
```
Comment (awly) [inline projects/teleport/build.sh:18]: we have some extra fun symlinking to do with vendored modules
```suggestion
make update-vendor
```

Comment (catenacyber) [reply projects/teleport/build.sh]: Thanks, it worked already with `go mod tidy && go mod vendor` but I made the suggested change

**Comment by catenacyber** [comment]
> Can you check the failures

So, here is a sum-up :
`compile_go_fuzzer` runs `go mod tidy` because certain project need it (they do not have go.sum even if they have go.mod)
This breaks go-fuzz for syzkaller
`go mod vendor` fixes go-fuzz for syzkaller even if it fails itself (ie `go mod vendor` returns 1)

**Pull request merged**